jetstream/maxtext/single-host-inference/README.md

```diff
@@ -0,0 +1,157 @@
+```
+gcloud config set project [PROJECT_ID]
+export PROJECT_ID=$(gcloud config get project)
+export REGION=[COMPUTE_REGION]
+export ZONE=[ZONE]
+```
+
+### Create GKE cluster and node pool
+```
+# Create zonal cluster with 2 CPU nodes
+gcloud container clusters create jetstream-maxtext \
+    --zone=${ZONE} \
+    --project=${PROJECT_ID} \
+    --workload-pool=${PROJECT_ID}.svc.id.goog \
+    --release-channel=rapid \
+    --num-nodes=2
+
+# Create one v5e TPU pool with topology 2x4 (1 TPU node with 8 chips)
+gcloud container node-pools create tpu \
+    --cluster=jetstream-maxtext \
+    --zone=${ZONE} \
+    --num-nodes=2 \
+    --machine-type=ct5lp-hightpu-8t \
+    --project=${PROJECT_ID}
+```
+You have created the following resources:
+
+- Standard cluster with 2 CPU nodes.
+- One v5e TPU node pool with 2 nodes, each with 8 chips.
+
+### Configure Applications to use Workload Identity
+Prequisite: make sure you have the following roles
+
+```
+roles/container.admin
+roles/iam.serviceAccountAdmin
+```
+
+Follow [these steps](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to) to configure the IAM and Kubernetes service account:
+
+```
+# Get credentials for your cluster
+$ gcloud container clusters get-credentials jetstream-maxtext \
+    --region=${REGION}
+
+# Create an IAM service account.
+$ gcloud iam service-accounts create jetstream-iam-sa
+
+# Ensure the IAM service account has necessary roles. Here we add roles/storage.objectUser for gcs bucket access.
+$ gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:jetstream-iam-sa@${PROJECT_ID}.iam.gserviceaccount.com" \
+    --role roles/storage.objectUser
+
+$ gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:jetstream-iam-sa@${PROJECT_ID}.iam.gserviceaccount.com" \
+    --role roles/storage.insightsCollectorService
+
+# Allow the Kubernetes default service account to impersonate the IAM service account
+$ gcloud iam service-accounts add-iam-policy-binding jetstream-iam-sa@${PROJECT_ID}.iam.gserviceaccount.com \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[default/default]"
+
+# Annotate the Kubernetes service account with the email address of the IAM service account.
+$ kubectl annotate serviceaccount default \
+    iam.gke.io/gcp-service-account=jetstream-iam-sa@${PROJECT_ID}.iam.gserviceaccount.com
+```
+
+## Deploy Maxengine Server and HTTP Server
+
+You can use the provided Maxengine server and HTTP server images already in `deployment.yaml` or [build your own](#optionals)
+
+Add desired overrides to your yaml file by editing the `args` in `deployment.yaml`. You can reference the [MaxText base config file](https://github.com/google/maxtext/blob/main/MaxText/configs/base.yml) on what values can be overridden.
+
+Use a checkpoint by adding `load_parameters_path=<GCS bucket path to your checkpoint>` under `args`
```

Comment:
"Configure the model checkpoint by adding..."

Reply:
"Or you could try deploy the sample yaml without adding the checkpoint path. In the `deployment.yaml`, we will deploy a Maxengine server targeting  Gemma-7b model" I feel we may add slightly more description about what model we are deploying and its characteristics so that user may understand the args easier, wdyt?